include/service/jobs/sites/site_create_edit_job.js

```diff
@@ -96,7 +93,24 @@ module.exports = function SiteCreateEditJobModule(pb) {
                         site = pb.DocumentCreator.create('site', mySite);
                     }
 
-                    site.hostname = mySite.hostname || site.hostname;
+                    var prevHostname = site.hostname;
+                    var newHostname = mySite.hostname;
+                    if (!site.prevHostnames) {
+                        site.prevHostnames = [];
```

Comment:
Move this if check and assignment to the ```SiteService.format``` method.  On save a number of methods and events get called:

1. The format event is fired.  This event allows for raw DTOs to be properly handled.  i.e. ISO-8601 strings to Date objects, and sanitizing string fields.
2. The object is retrieved from the DB (only when ID is set)
3. The merge event is fired.  This is where the properties from the DTO are merged into the existing entity object.
4. The validate event is fired.  This allows for the whole object to be validated before being persisted.  When errors occur they are handed back in the callback as an error with a ```validationErrors``` property.
5. The before save event is fired. This is there just in case there is a last minute action to perform before persistence.
6. The object is persisted to the DB
7. The after save event is fired.  This is here in case you want to do caching or something along those lines.